File: ClipMobile/src/main/java/com/migueljteixeira/clipmobile/ui/BaseFragment.java
package com.migueljteixeira.clipmobile.ui;

import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.FrameLayout;

import androidx.annotation.Nullable;
import androidx.cardview.widget.CardView;
import androidx.fragment.app.Fragment;

import com.migueljteixeira.clipmobile.R;

import butterknife.Bind;
import butterknife.ButterKnife;

public class BaseFragment extends Fragment {

    @Nullable
    @Bind(R.id.progress_spinner)
    FrameLayout mProgressSpinner;
    @Nullable
    @Bind(R.id.main_view)
    CardView mMainView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Retain this fragment across configuration changes.
        setRetainInstance(true);
    }

    /**
     * Shows the progress spinner and hides the login form.
     */
    protected void showProgressSpinner(final boolean show) {
        mProgressSpinner.setVisibility(show ? View.VISIBLE : View.GONE);
        mMainView.setVisibility(show ? View.GONE : View.VISIBLE);
    }

    /**
     * Shows the progress spinner
     */
    protected void showProgressSpinnerOnly(final boolean show) {
        mProgressSpinner.setVisibility(show ? View.VISIBLE : View.GONE);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();

        ButterKnife.unbind(this);
    }

    protected void cancelTasks(AsyncTask mTask) {
        if (mTask != null && mTask.getStatus() != AsyncTask.Status.FINISHED)
            mTask.cancel(true);
    }

}

File: ClipMobile/src/main/java/com/migueljteixeira/clipmobile/ui/BaseFragment.kt
package com.migueljteixeira.clipmobile.ui

import android.widget.FrameLayout
import androidx.cardview.widget.CardView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.migueljteixeira.clipmobile.R
import android.os.AsyncTask
import android.view.View
import androidx.fragment.app.Fragment

open class BaseFragment : Fragment() {
    var mProgressSpinner: FrameLayout? = null
    var mMainView: CardView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val root = super.onCreateView(inflater, container, savedInstanceState)!!
        mProgressSpinner = root.findViewById(R.id.progress_spinner)
        mMainView = root.findViewById(R.id.main_view)
        return root
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Retain this fragment across configuration changes.
        retainInstance = true
    }

    fun bindHelperViews(view: View?) {
        var view = view
        if (view == null) view = getView()
        assert(view != null)
        mProgressSpinner = view!!.findViewById(R.id.progress_spinner)
        mMainView = view.findViewById(R.id.main_view)
    }

    /**
     * Shows the progress spinner and hides the login form.
     */
    protected fun showProgressSpinner(show: Boolean) {
        mProgressSpinner?.visibility = if (show) View.VISIBLE else View.GONE
        mMainView?.visibility = if (show) View.GONE else View.VISIBLE
    }

    /**
     * Shows the progress spinner
     */
    protected fun showProgressSpinnerOnly(show: Boolean) {
        mProgressSpinner?.visibility = if (show) View.VISIBLE else View.GONE
    }

    protected fun cancelTasks(mTask: AsyncTask<*, *, *>?) {
        if (mTask != null && mTask.status != AsyncTask.Status.FINISHED) mTask.cancel(true)
    }
}